src/hex/net/serializer.rs

```diff
@@ -0,0 +1,54 @@
+use std::fs;
+
+use crate::game::common::{GameColor, GamePosition};
+use crate::game::self_play::DataSerializer;
+use crate::hex::hex_game::{HexGame, HexMove, HexPosition, BOARD_SIZE};
+use crate::hex::net;
+
+pub struct HexSerializer {}
+
+impl HexSerializer {
+    pub fn new() -> Self {
+        Self {}
+    }
+}
+
+impl DataSerializer<HexGame> for HexSerializer {
+    fn serialize_data_entry_to_file(
+        &self,
+        pos: HexPosition,
+        probs: Vec<(HexMove, f32)>,
+        winner: Option<GameColor>,
+        filename: String,
+    ) -> std::io::Result<()> {
+        /* Always serialize as turn=1 */

```

Comment:
This is kinda weird that the `Serializer` does the flipping (as opposed to its caller function). Especially because we want this similar behavior for all of the games, so this code would be duplicated. Not critical to change but still if you have an elegant way to do this I think it could be less confusing

Reply:
i agree, to change that we need to abstract the flipping of positions and moves probabilities, ill open a ticket